cmake/options.cmake

```diff
@@ -0,0 +1,84 @@
+option(PRESSIO_ENABLE_CXX14 "Enable C++14" OFF)
```

Comment:
why is this file inside cmake/ ? the source code does not need this at all only the tests need this. 
so i would move this inside the tests subdir

Reply:
Moved to `tests` subdir.